### PR TITLE
[tests-only]added tests for public link

### PR DESCRIPTION
**Pull request by swoichha**
### Description
This PR adds tests for sharing file using public link.

### Related Issue
https://github.com/owncloud/client/issues/8501

### CODE REVIEW

File: test/gui/shared/steps/steps.py [hunk 1]
```diff
@@ -233,6 +233,13 @@ def step(context, stepPart1, stepPart2):
     )
 
 
+@Then(r"^(.*) on the server", regexp=True)
```
Comment (dpakach) [inline test/gui/shared/steps/steps.py:236]: I'm not fully sure, but you may make this work with any step with `on the server` in it
```suggestion
@Then(r"^(.*)on the server(.*)", regexp=True)
```

File: test/gui/shared/steps/steps.py [hunk 2]
```diff
@@ -362,4 +369,16 @@ def step(context, receiver, resource):
...
+ 
+@When('the user creates a new public link for file "|any|" without password using the client-UI')
+def step(context, resource):
+    test.compare(str(waitForObjectExists(names.sharingDialog_label_name_QLabel).text), resource)
+    clickButton(waitForObject(names.oCC_ShareLinkWidget_createShareButton_QPushButton))     
+    
+@When('the user creates a new public link for file "|any|" with password "|any|" using the client-UI')
+def step(context, resource, password):
+    test.compare(str(waitForObjectExists(names.sharingDialog_label_name_QLabel).text), resource)
+    snooze(10)
```
Comment (dpakach) [inline test/gui/shared/steps/steps.py:381]: why is the sleep here?

Comment (swoichha) [reply test/gui/shared/steps/steps.py:381]: That was added to get object for password field but we won't be needing it now. I will remove it in my next commit.

File: test/gui/shared/steps/steps.py (hunk 1 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:236]: This way doesn't seem to work. So, I think we should keep this as it is.

**Review by dpakach — COMMENTED** [review]
lgtm :+1:

File: test/gui/tst_sharing/test.feature [hunk 3]
```diff
@@ -87,3 +87,56 @@ Feature: Sharing
...
+            0\display-name=Alice
+            0\http_oauth=false
+            0\http_user=alice
+            0\url=%local_server%
+            0\user=Alice
+            0\version=1
+            version=2
+            """
+        When the user opens the public links dialog of "%client_sync_path%/textfile0.txt" using the client-UI
+        And the user creates a new public link for file "textfile0.txt" without password using the client-UI
```
Comment (individual-it) [inline test/gui/tst_sharing/test.feature:113]: I think this step should do the complete action. Opening the dialog + sharing

File: test/gui/shared/steps/steps.py [hunk 4]
```diff
@@ -350,16 +357,27 @@ def step(context):
...
 @Then('the error text "|any|" should be displayed in the sharing dialog')
 def step(context, fileShareContext):
     test.compare(str(waitForObjectExists(names.sharingDialog_The_file_can_not_be_shared_because_it_was_shared_without_sharing_permission_QLabel).text), fileShareContext)
 
-                 
\ No newline at end of file
+
+@When('the user creates a new public link for file "|any|" without password using the client-UI')
+def step(context, resource):
+    test.compare(str(waitForObjectExists(names.sharingDialog_label_name_QLabel).text), resource)
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:375]: we should first wait till the file is synced, in case its not yet

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/tst_sharing/test.feature [hunk 5]
```diff
@@ -87,3 +87,55 @@ Feature: Sharing
...
+            0\http_oauth=false
+            0\http_user=alice
+            0\url=%local_server%
+            0\user=Alice
+            0\version=1
+            version=2
+            """
+        When the user opens the public links dialog of "%client_sync_path%/textfile0.txt" using the client-UI
+        And the user toggles the password protection using the client-UI
+        And the user creates a new public link for file "textfile0.txt" with password "pass123" using the client-UI
```
Comment (individual-it) [inline test/gui/tst_sharing/test.feature:139]: same here, lets have a combined high-level step that does it all

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/shared/steps/steps.py [hunk 6]
```diff
@@ -350,16 +357,38 @@ def step(context):
...
\ No newline at end of file
+
+@When('the user creates a new public link for file "|any|" without password using the client-UI')
+def step(context, resource):
+    resource = substituteInLineCodes(context, resource).replace('//','/')
+    waitFor(lambda: isFileSynced(resource), context.userData['clientSyncTimeout'] * 1000)
+    waitFor(lambda: shareResource(resource), context.userData['clientSyncTimeout'] * 1000)
+    mouseClick(waitForObject(names.qt_tabwidget_tabbar_Public_Links_TabItem), 0, 0, Qt.NoModifier, Qt.LeftButton)
+    
+    test.compare(str(waitForObjectExists(names.sharingDialog_label_name_QLabel).text), resource.replace(context.userData['clientSyncPath'], ''))
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:380]: duplicated code

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**